File: 27-Remove-Element.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int i, j = nums.size() - 1;
        int k = 0;

        if (j == 0 && nums[0] == val) return 0;

        for (i = 0; i < nums.size()  && i < j; i++) {
            if (nums[i] == val) {
                while (j > i && nums[j] == val) j--;
                if (j == i && nums[j] == val) break;
                int temp = nums[i];
                nums[i] = nums[j];
                nums[j] = temp;
                j--;
            }
            k++;
        }
        if (i == j && nums[j] != val) k++;

        return k;
        
    }
};

File: 27-Remove-Element.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int l = 0;
        int r = nums.length - 1;
    
        while (l <= r) {
            if (nums[l] == val) {
                nums[l] = nums[r];
                r--;
            } else {
                l++;
            }
        }

        return r + 1;
    }
}